File: SakhaTyla.Web.Ng/src/app/file-group-core/file-group.model.ts
import { PageSettings } from '../core/page.model';
import { FileGroupFilter } from './file-group-filter.model';
import { FileGroupType } from '../file-group-type/file-group-type.model';

export class FileGroup {
  id: number;
  creationDate: Date;
  modificationDate: Date;
  creationUserId: number;
  modificationUserId: number;
  name: string;
  type: FileGroupType;
  location: string;
  accept: string;
}

export class FileGroupListState {
  pageSize = PageSettings.pageSize;
  pageIndex = 0;
  filter = new FileGroupFilter();
}

File: SakhaTyla.Core/Requests/FileGroups/Models/FileGroupModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace SakhaTyla.Core.Requests.FileGroups.Models
{
    public class FileGroupModel
    {
        [DisplayName("Id")]
        public int Id { get; set; }

        [DisplayName("Creation Date")]
        public DateTime CreationDate { get; set; }

        [DisplayName("Modification Date")]
        public DateTime ModificationDate { get; set; }

        [DisplayName("Creation User")]
        public int? CreationUserId { get; set; }
        public Users.Models.UserShortModel CreationUser { get; set; }

        [DisplayName("Modification User")]
        public int? ModificationUserId { get; set; }
        public Users.Models.UserShortModel ModificationUser { get; set; }

        [DisplayName("Name")]
        public string Name { get; set; }

        [DisplayName("Type")]
        public Enums.FileGroupType Type { get; set; }

        [DisplayName("Location")]
        public string Location { get; set; }

        [DisplayName("Accept")]
        public string Accept { get; set; }
    }
}
